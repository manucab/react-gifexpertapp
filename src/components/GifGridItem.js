import React from 'react'

const GifGridItem = ( {id, title, url} ) => {
    

    return (
        <div className='card animate_animated animate__fadeIn'>
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

export default GifGridItem
