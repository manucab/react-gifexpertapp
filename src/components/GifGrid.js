import React from "react";
import { useFetchGifs } from "../Hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate_animated animate__fadeIn'> {category} </h3>

      {loading && <p className='animate_animated animate__flash'>Cargando</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
