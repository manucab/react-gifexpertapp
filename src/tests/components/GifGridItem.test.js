
import React from 'react';
import {GifGridItem} from '../../components/GifGridItem'
import {shallow} from 'enzyme'

describe('Pruebas en GifGridItem', () => {

    const title = 'Un t√≠tulo';
    const url = 'https://www.google.es';

    let wrapper = shallow(<GifGridItem title={title} url={url}/>);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />)
    })

    
    test('debe mostrar el componente correctamente', () => {
        
      expect( wrapper ).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );

    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
      
        const img = wrapper.find('img');

        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );

    });

    test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className ).toContain( 'animate__fadeIn' )

    });


    
    
    
    

})
