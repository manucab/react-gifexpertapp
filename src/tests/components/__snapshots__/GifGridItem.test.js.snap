// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate_animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://www.google.es"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
